src/bundle/ui-dev/src/modules/universal-discovery/bookmarks.tab.module.js

```diff
@@ -41,11 +42,24 @@ const BookmarksTabModule = () => {
         return views[currentView];
     };
 
+    useEffect(() => {
+        setMarkedLocationId(null);
+        dispatchLoadedLocationsAction({ type: 'CLEAR_LOCATIONS' });
+
+        return () => {
+            if (shouldRestorePreviousStateRef.current) {
+                setMarkedLocationId(markedLocationId);
```

Comment:
I think it will always restore the location which I enter the bookmarks. It should keep the selected location on leaving the bookmarks tab if I selected some location from bookmark list.